cumulus_library/template_sql/templates.py

```diff
@@ -194,24 +194,37 @@ class CodeableConceptConfig:
 
     :param source_table: the table to extract extensions from
     :param source_id: the id field to use in the new table
-    :param cc_columns: the column containing the codeableConcept you want to extract.
-        Format:
-            {'name':[column],
-            'is_array': [boolean],
-            'code_systems':[List of code system strings, in priority order]}
-        is_array relates to the FHIR spec - if the field is specified
-        as 0...*, set this to be true.
+    :param column_name: the column containing the codeableConcept you want to extract.
+    :param is_array: whether the codeableConcept is 0...1 or 0..* in the FHIR spec
     :param target_table: the name of the table to create
+    :param filter_priority: If true, will use code systems to select a single code,
+      in preference order, for use as a display value.
     :param code_systems: a list of systems, in preference order, for selecting data
+      for filtering. This should not be set if filter_priority is false.
     """
 
     def __init__(
-        self, source_table: str, source_id: str, cc_column: dict, target_table: str
+        self,
+        source_table: str,
+        source_id: str,
+        column_name: str,
+        is_array: bool,
+        target_table: str,
+        filter_priority: bool = False,
+        code_systems: list = ["all"],
```

Comment:
nit: I would strongly caution against any container as a default argument. It's only a _problem_ if you then try to modify that container, which isn't always the case (and doesn't seem to be here?). But it's a lingering rake to trip over.

(The problem is that Python creates the default argument _once_ and re-uses it. So any modifications to `code_systems` will carry over to future calls.)

I usually prefer to use `None` for this sort of thing, then in the body, do something like `code_systems = code_systems or ["all"]` to get a similar result without the risk and with only one line of fuss.

Reply:
I had a similar thought about this really needing to be none - this ended up coming down to me being lazy w.r.t. how the jinja template is updated, and i just covid-brained shoved it here.

I do think there's some DRY repercussions in the template, but I can moved this sort of proxy for 'i need the for loop in the template to run exactly once' down into the function that uses this config on a 'is None' detection.

Reply:
maybe not as much as i really thought - i'll just do it the right way rather than trying to hack around it.